api.planx.uk/modules/analytics/metabase/shared/client.test.ts

```diff
@@ -1 +1,127 @@
-test.todo("should test configuration and errors");
+import axios from "axios";
+import {
+  validateConfig,
+  createMetabaseClient,
+  MetabaseError,
+  initializeMetabaseClient,
+} from "./client.js";
+import nock from "nock";
+
+const axiosCreateSpy = vi.spyOn(axios, "create");
+
+describe("Metabase client", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.resetModules();
+    process.env.METABASE_URL_EXT = "https://test-metabase-url.com";
+    process.env.METABASE_API_KEY = "test-api-key";
+    initializeMetabaseClient();
+  });
+
+  afterEach(() => {
+    vi.unstubAllEnvs();
+  });
+
+  test("returns configured client", async () => {
+    const _client = createMetabaseClient();
+
+    expect(axiosCreateSpy).toHaveBeenCalledWith({
+      baseURL: process.env.METABASE_URL_EXT,
+      headers: {
+        "X-API-Key": process.env.METABASE_API_KEY,
+        "Content-Type": "application/json",
+      },
+      timeout: 30_000,
+    });
```

Comment:
I think we can improve this to make this test more meaningful - currently we're not actually testing the returned client. Technically `createMetabaseClient()` could be calling axios (so this test passes), but then returning something invalid for the client.

We can test the returned client like this, for each configuration we've customised - 

```ts
expect(client.defaults.headers["X-API-Key"]).toBe(process.env.METABASE_API_KEY);
```

This then actually tests we're getting back what we expected 👍